relentless/potential/potential.py

```diff
@@ -294,48 +312,91 @@ def __init__(self, types, params, default={}):
 
         self.coeff = CoefficientMatrix(types, params, default)
 
-        self.id = PairPotential._id
-        PairPotential._id += 1
-
     def energy(self, pair, r):
-        """ Evaluate energy for a (i,j) pair."""
+        """Evaluate energy for a (i,j) pair.
+
+        Parameters
+        ----------
+        pair : array_like
+            The pair for which to calculate the energy.
+        r : array_like
+            The location(s) at which to evaluate the energy.
+
+        Returns
+        -------
+        array_like
+            The energy at the specified location(s).
+
+        Raises
+        ------
+        ValueError
+            If any value in `r` is negative.
+        ValueError
+            If the potential is to be shifted without setting `rmax`.
+
+        """
+        if np.min(r) < 0:
+            raise ValueError('r cannot be negative')
         params = self.coeff.evaluate(pair)
         r,u,scalar_r = self._zeros(r)
```

Comment:
I would move the check on r here, and just use:
```py
any(r < 0)
```
since it is easier to read.

You can move the `evaluate` after this check, if you only want to do this after you do the error checking.